### PR TITLE
feat(metrics): Add a `flow.signup.link.signin` flow event.

### LINKED ISSUE #4680: Metrics are not guaranteed to be flushed when user clicks "sign in" from /connect_another_device
Discovered while chatting with @philbooth about https://github.com/mozilla/fxa-content-server/pull/4676.

The "Sign in" button from Connect Another Device immediately redirects the user to the /signin page w/o first flushing metrics. This works great on most browsers that support sending XHR requests in the `beforeunload` handler, which is everybody we target right now. But, if the user has sendBeacon disabled, metrics won't be flushed.

We should ensure metrics are flushed before redirecting the user.

### LINKED ISSUE #5115: Clicks on "Looking for Firefox Sync? Get started here" are not logged as flow events.
We have similar logic in [connect_another_device](https://github.com/mozilla/fxa-content-server/blob/277e046a3a27ff946a709fd03f10e0fa04ad565c/app/scripts/templates/connect_another_device.mustache#L19) that logs a `link.signin` flow event.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
fixes #4680
fixes #5115 

@mozilla/fxa-devs - r?

### CODE REVIEW

File: app/tests/spec/views/sign_up.js [hunk 1]
```diff
@@ -1341,6 +1341,15 @@ define(function (require, exports, module) {
         view.$('#submit-btn').click();
         assert.isTrue(TestHelpers.isEventLogged(metrics, 'flow.signup.submit'));
       });
+
+      it('logs the suggest-sync event', () => {
```
Comment (philbooth) [inline app/tests/spec/views/sign_up.js]: The description seems wrong. Copy/paste?

Comment (shane-tomlinson) [inline app/tests/spec/views/sign_up.js]: Thanks for catching that, I erroneously put the element ID instead if the event name, will update.

**Comment by philbooth** [comment]
r+

**Comment by vladikoff** [comment]
```
Failed test: focuses the email element by default
 => timeout of 20000ms exceeded
Runnable.prototype.resetTimeout/this.timer<@http://127.0.0.1:3030/bower_components/mocha/mocha.js:4254:19
× firefox on any platform - mocha tests - run the mocha tests (104.904s)
Error: Expected 0 mocha test failures
```

failed in both travis runs, related?

**Comment by shane-tomlinson** [comment]
> failed in both travis runs, related?

I'm sure of it!

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: ceb9cafb5d09d21ec6b0b6a569c21d5d8f2dcb5c
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: ceb9cafb5d09d21ec6b0b6a569c21d5d8f2dcb5c
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Pull request merged**